openmdao_book/features/core_features/working_with_derivatives/parallel_derivs.ipynb

```diff
@@ -0,0 +1,319 @@
+    "        super().__init__()\n",
+    "        self.delay = delay\n",
+    "        self.size = size\n",
+    "        self.mult = mult\n",
+    "\n",
+    "    def setup(self):\n",
+    "        self.add_input('x', val=0.0)\n",
+    "        self.add_output('y', val=np.zeros(self.size))\n",
+    "\n",
+    "        self.declare_partials(of='*', wrt='*')\n",
+    "\n",
+    "    def compute(self, inputs, outputs):\n",
+    "        outputs['y'] = inputs['x'] * self.mult\n",
+    "\n",
+    "    def compute_partials(self, inputs, partials):\n",
+    "        partials['y', 'x'] = self.mult\n",
+    "\n",
+    "    def _apply_linear(self, jac, vec_names, rel_systems, mode, scope_out=None, scope_in=None):\n",
+    "        time.sleep(self.delay)\n",
+    "        super()._apply_linear(jac, vec_names, rel_systems, mode, scope_out, scope_in)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "db94b22e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class PartialDependGroup(om.Group):\n",
+    "    def setup(self):\n",
+    "        size = 4\n",
+    "\n",
+    "        Comp1 = self.add_subsystem('Comp1', SumComp(size))\n",
+    "        pargroup = self.add_subsystem('ParallelGroup1', om.ParallelGroup())\n",
+    "\n",
+    "        self.set_input_defaults('Comp1.x', val=np.arange(size, dtype=float)+1.0)\n",
+    "\n",
+    "        self.linear_solver = om.LinearBlockGS()\n",
+    "        self.linear_solver.options['iprint'] = -1\n",
+    "        pargroup.linear_solver = om.LinearBlockGS()\n",
+    "        pargroup.linear_solver.options['iprint'] = -1\n",
+    "\n",
+    "        delay = .1\n",
+    "        Con1 = pargroup.add_subsystem('Con1', SlowComp(delay=delay, size=2, mult=2.0))\n",
+    "        Con2 = pargroup.add_subsystem('Con2', SlowComp(delay=delay, size=2, mult=-3.0))\n",
+    "\n",
+    "        self.connect('Comp1.y', 'ParallelGroup1.Con1.x')\n",
+    "        self.connect('Comp1.y', 'ParallelGroup1.Con2.x')\n",
+    "\n",
+    "        color = 'parcon'\n",
+    "        self.add_design_var('Comp1.x')\n",
+    "        self.add_constraint('ParallelGroup1.Con1.y', lower=0.0, parallel_deriv_color=color)\n",
+    "        self.add_constraint('ParallelGroup1.Con2.y', upper=0.0, parallel_deriv_color=color)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a7dbbfb0",
+   "metadata": {},
+   "source": [
+    "### Run script"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "870c957c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!echo $PYTHONPATH"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "be875778",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!echo $PATH"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0e3dcabd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%env DYLD_LIBRARY_PATH=/Users/hschilli/ipopt/lib:/opt/openmpi/lib"
```

Comment:
hardwired path to '/Users/hschilli/ipopt/lib:/opt/openmpi/lib' (multiple places)

Reply:
Argh. I meant to take all those out. That was just there for debugging.